backends/candle/src/lib.rs

```diff
@@ -673,3 +568,45 @@ impl<O> WrapErr<O> for Result<O, candle::Error> {
         self.map_err(|e| BackendError::Inference(e.to_string()))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use text_embeddings_backend_core::Batch;
+
+    #[test]
+    #[should_panic(expected = "Invalid cumulative sequence lengths")]
+    fn test_invalid_cumulative_seq_lengths() {
+        // Create a mock backend for testing
+        let device = Device::Cpu;
+        let model = Box::new(MockModel);
+        let backend = CandleBackend { device, model };
+
+        // Create a batch with invalid cumulative sequence lengths (decreasing)
+        let batch = Batch {
+            input_ids: vec![1, 2, 3, 4],
+            token_type_ids: vec![0, 0, 0, 0],
+            position_ids: vec![0, 1, 2, 3],
+            cumulative_seq_lengths: vec![0, 3, 2], // Invalid: 3 > 2
```

Comment:
i think this is impossible by convention. the tensor is named cumulative_seq_lengths. Theoretically, you could also overflow the input_ids by exceeding u32 or u64, if you were to train a model. 

Please close the PR.